clarify native package (#8576)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/packages.md

### Native private packages <Lifecycle status='beta'/> 

Native private packages let you install packages from [supported](#prerequisites) private <Constant name="git" /> repos using the `private` key, without having to configure a [token](#git-token-method) or write out a full Git URL. 

- <Constant name="dbt_platform" /> &mdash; Uses your existing <Constant name="git" /> [integration](/docs/cloud/git/git-configuration-in-dbt-cloud) for authentication.
- <Constant name="fusion" /> locally &mdash; Uses your system's SSH configuration. Requires the [`provider` key](#using-the-provider-key).

#### Prerequisites

- You must have the feature flag enabled. Contact your account team to request access.

> - **<Constant name="dbt_platform" />** &mdash; You must have one of the following <Constant name="git" /> providers configured in the **Integrations** section of your **Account settings**:

Author: ```suggestion
- To use native private packages, you must have one of the following <Constant name="git" /> providers configured in the **Integrations** section of your **Account settings**:
```

  - [GitHub](/docs/cloud/git/connect-github)
  - [Azure DevOps](/docs/cloud/git/connect-azure-devops)
    - Private packages only work within a single Azure DevOps project. If your repositories are in different projects within the same organization, you can't reference them in the `private` key at this time.
    - For Azure DevOps, use the `org/repo` path (not the `org_name/project_name/repo_name` path) with the project tier inherited from the integrated source repository.
  - [GitLab](/docs/cloud/git/connect-gitlab)
    - Every GitLab repo with private packages must also be a <Constant name="cloud" /> project. 

> - **<Constant name="fusion" /> locally** &mdash; You must have an SSH key configured on your machine for the relevant Git provider and include the [`provider` key](#using-the-provider-key) in your package configuration.

Author: ```suggestion
- If using <Constant name="fusion" /> locally, you must have an SSH key configured on your machine for the relevant Git provider and include the [`provider` key](#using-the-provider-key) in your package configuration.
```

### Native private packages <Lifecycle status='beta'/> 

> Native private packages let you install packages from [supported](#prerequisites) private <Constant name="git" /> repos using the `private` key, without having to configure a [token](#git-token-method) or write out a full Git URL. 

Reviewer: Maybe a question for @eliasdefaria: what is the benefit of private key vs token plus full git URL? Simplified authentication, easier set up and maintenance?? 

Maybe we can call this out in a sentence.

Author: addressed https://github.com/dbt-labs/docs.getdbt.com/pull/8576/changes#r2792246146

> Native private packages let you install packages from [supported](#prerequisites) private <Constant name="git" /> repos using the `private` key, without having to configure a [token](#git-token-method) or write out a full Git URL. 

Author: ### 
```suggestion
Native private packages let you install packages from [supported](#prerequisites) private <Constant name="git" /> repos using the `private` key, without having to configure a [token](#git-token-method) or write out a full Git URL. This simplifies setup and reduces credential management.
```

> - <Constant name="dbt_platform" /> &mdash; Uses your existing <Constant name="git" /> [integration](/docs/cloud/git/git-configuration-in-dbt-cloud) for authentication.

Author: ```suggestion
- <Constant name="dbt_platform" /> &mdash; Uses your existing <Constant name="git" /> [integration](/docs/cloud/git/git-configuration-in-dbt-cloud) for authentication. Needs the `private` key.
```

> - <Constant name="dbt_platform" /> &mdash; Uses your existing <Constant name="git" /> [integration](/docs/cloud/git/git-configuration-in-dbt-cloud) for authentication. Needs the `private` key.

Author: ```suggestion
- <Constant name="dbt_platform" /> &mdash; Uses your existing <Constant name="git" /> [integration](/docs/cloud/git/git-configuration-in-dbt-cloud) for authentication.
```

> - <Constant name="dbt_platform" /> &mdash; Uses your existing <Constant name="git" /> [integration](/docs/cloud/git/git-configuration-in-dbt-cloud) for authentication. Needs the `private` key.

Reviewer: ```suggestion
- <Constant name="dbt_platform" />: Uses your existing <Constant name="git" /> [integration](/docs/cloud/git/git-configuration-in-dbt-cloud) for authentication. Needs the `private` key.
```

> - <Constant name="dbt_platform" /> &mdash; Uses your existing <Constant name="git" /> [integration](/docs/cloud/git/git-configuration-in-dbt-cloud) for authentication.

Reviewer: ```suggestion
- <Constant name="dbt_platform" />: Uses your existing <Constant name="git" /> [integration](/docs/cloud/git/git-configuration-in-dbt-cloud) for authentication.
```

> - <Constant name="fusion" /> locally &mdash; Uses your system's SSH configuration. Requires the [`provider` key](#using-the-provider-key).

Reviewer: ```suggestion
- <Constant name="fusion" /> locally: Uses your system's SSH configuration. Requires the [`provider` key](#using-the-provider-key).
```

  - [Azure DevOps](/docs/cloud/git/connect-azure-devops)
    - Private packages only work within a single Azure DevOps project. If your repositories are in different projects within the same organization, you can't reference them in the `private` key at this time.
    - For Azure DevOps, use the `org/repo` path (not the `org_name/project_name/repo_name` path) with the project tier inherited from the integrated source repository.
  - [GitLab](/docs/cloud/git/connect-gitlab)
    - Every GitLab repo with private packages must also be a <Constant name="cloud" /> project. 
- If using <Constant name="fusion" /> locally, you must have an SSH key configured on your machine for the relevant Git provider and include the [`provider` key](#using-the-provider-key) in your package configuration.

#### Configuration

> Use the `private` key in your `packages.yml` or `dependencies.yml` to clone package repos without having to provision an access token or manage a Git URL.

Author: ```suggestion
Use the `private` key in your `packages.yml` or `dependencies.yml` to clone package repos using your existing <Constant name="cloud" /> Git integration without having to provision an access token or create a <Constant name="cloud" /> environment variable. 
```

  
```

#### Using the `provider` key

> If you are using multiple <Constant name="git" /> integrations or using the <Constant name="fusion_engine" />, add the `provider` key. When using <Constant name="fusion" /> locally (with the [<Constant name="fusion" /> CLI](/docs/fusion/install-fusion-cli) or the [VS Code extension](/docs/fusion/install-dbt-extension)), the `provider` key is **required**.

Author: ```suggestion
Add the `provider` key when:
- You are using multiple <Constant name="git" /> integrations or using the <Constant name="fusion_engine" />.
- You are using <Constant name="fusion" /> locally (with the [<Constant name="fusion" /> CLI](/docs/fusion/install-fusion-cli) or the [VS Code extension](/docs/fusion/install-dbt-extension)) (required).
```

  
```

#### Using the `provider` key

Add the `provider` key when:
- You are using multiple <Constant name="git" /> integrations or using the <Constant name="fusion_engine" />.
- You are using <Constant name="fusion" /> locally (with the [<Constant name="fusion" /> CLI](/docs/fusion/install-fusion-cli) or the [VS Code extension](/docs/fusion/install-dbt-extension)) (required).

```yaml
packages:
  - private: dbt-labs/awesome_repo
    provider: "github" # Supported values: "github", "gitlab", "azure_devops"
```

<Constant name="fusion" /> uses the `provider` value to construct the correct SSH URL for cloning, based on the provider:

> 

Reviewer: This is awesome and super helpful!

| Provider | SSH URL format |
| --- | --- |
| `github` | `git@github.com:org/repo.git` |
| `gitlab` | `git@gitlab.com:org/repo.git` |
| `azure_devops` | `git@ssh.dev.azure.com:v3/org/repo` |

<Constant name="fusion" /> relies on your system's SSH configuration to authenticate and clone the private repository. If `git clone` works on your system for the private package repo, the private package install should work too.

### SSH key method (CLI only)

:::note

> This method uses the `git:` key with a full SSH URL, which is different from [native private packages](#native-private-packages) that use the `private:` key. For most use cases, native private packages is the recommended approach.

Reviewer: Also adding the why here would be great!

> This method uses the `git:` key with a full SSH URL, which is different from [native private packages](#native-private-packages) that use the `private:` key. For most use cases, native private packages is the recommended approach.

Author: ```suggestion
This method uses the `git:` key with a full SSH URL, which is different from [native private packages](#native-private-packages) that use the `private:` key. For most use cases, native private packages is the recommended approach as it simplifies setup.
```
